### ISSUE TITLE
Zarr format: Allow writing attributes as `int64`

### ISSUE THREAD

**Issue by eschnett** [OSGeo/gdal#10062]
### Feature description

The Zarr file format automatically converts 64-bit integer attributes into 64-bit floating-point attributes. I assume the reason is that attributes are stored as JSON and JSON doesn't guarantee that 64-bit integers can safely round-trip. However, reading 64-bit integer attributes is supported, probably in case another packages creates such an attribute.

I would like to have a way (e.g. an option when creating a file, or when creating a multidim array?) to write 64-bit integer attributes as integers in JSON, without converting them to a floating-point number.

### Additional context

If you agree this is a good idea I might be able to contribute such a change.

**Comment by rouault** [comment]
> ###
> 
> If you agree this is a good idea I might be able to contribute such a change.

Yes, we have all infrastructure needed for that. The changes should only be in frmts/zarr/zarr_attribute.cpp , and involve testing / using CPLJSONObject::Type::Long in the JSON -> GDAL side (towards line 167 and 189), and GDT_Int64 / GDT_UInt16 in the GDAL -> JSON side (lines 292 and 310). Please extend autotest/gdrivers/zarr_driver.py to test for that

**Comment by eschnett** [comment]
Implemented via https://github.com/OSGeo/gdal/pull/10065.
